lib/traject/solr_json_writer.rb

```diff
@@ -114,6 +114,12 @@ def initialize(argSettings)
       if @settings["solr_writer.http_timeout"]
         client.connect_timeout = client.receive_timeout = client.send_timeout = @settings["solr_writer.http_timeout"]
       end
+
+      if @settings["solr_writer.basic_auth_user"] &&
+          @settings["solr_writer.basic_auth_password"]
+        client.set_auth(@settings["solr.url"], @settings["solr_writer.basic_auth_user"], @settings["solr_writer.basic_auth_password"])
```

Comment:
Do you think it's less confusing if this check is `||` instead of `&&`? If someone accidentally sets only username or only password, which failure mode is least confusing? Or is it the same either way? are there any cases where you would on purpose set only the username but not the password, and that would work for auth?

Reply:
Either way the error will not get caught until we hit the solr server.  My inclination is to only set it once both username and password are set because the user will be able to determine that basic auth was not enabled on the server end vs. It was enabled but the password was missing (harder to determine).

Reply:
Oh, and I don't think there are any cases where one would say one but not the other.